core/services/log/registrations.go

```diff
@@ -110,31 +113,41 @@ func (r *registrations) removeSubscriber(reg registration) (needsResubscribe boo
 		}
 	}
 
-	r.resetHighestNumConfirmations()
+	r.resetNumConfirmationsRange(reg.opts.NumConfirmations)
 	return
 }
 
-// increase the highestNumConfirmations stored if the new listener has a higher value
-func (r *registrations) maybeIncreaseHighestNumConfirmations(newNumConfirmations uint64) {
+// maybe update the numbers tracking highest and lowest num confirmations
+func (r *registrations) maybeUpdateNumConfirmationsRange(newNumConfirmations uint64) {
 	if newNumConfirmations > r.highestNumConfirmations {
 		r.highestNumConfirmations = newNumConfirmations
 	}
+	if newNumConfirmations < r.lowestNumConfirmations {
+		r.lowestNumConfirmations = newNumConfirmations
+	}
 }
 
-// reset the highest confirmation number per all current listeners
-func (r *registrations) resetHighestNumConfirmations() {
-	highestNumConfirmations := uint64(0)
-
-	for _, perAddress := range r.registrations {
-		for _, perTopic := range perAddress {
-			for _, listener := range perTopic {
-				if listener.opts.NumConfirmations > highestNumConfirmations {
-					highestNumConfirmations = listener.opts.NumConfirmations
+// reset the numbers tracking highest and lowest num confirmations
+func (r *registrations) resetNumConfirmationsRange(removedNumConfirmations uint64) {
+	if removedNumConfirmations <= r.lowestNumConfirmations && removedNumConfirmations >= r.highestNumConfirmations {
```

Comment:
At first glance, this seems like a weird (/impossible) condition to check. Mind adding a comment?

Reply:
true, yes it can use `==` - otherwise it would be an inconsistent state anyway

Reply:
core/services/log/registrations.go#L70

SA4003: no value of type uint64 is less than 0 (staticcheck)

Reply:
good point!